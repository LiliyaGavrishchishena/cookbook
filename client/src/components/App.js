import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
// components

import RecipesPage from '../pages/RecipesPage';
import DetailPage from '../pages/DetailPage';
// configs
import routes from '../configs/routes';
// styles
import 'materialize-css';

const App = () => {
  return (
    <div className="container">
      <h1 className="hand blue-grey-text center-align">Cookbook</h1>
      <Switch>
        <Route exact path={routes.RECIPE} component={RecipesPage} />
        <Route exact path={routes.DETAIL} component={DetailPage} />

        <Redirect to={routes.RECIPE} />
      </Switch>
    </div>
  );
};

export default App;
