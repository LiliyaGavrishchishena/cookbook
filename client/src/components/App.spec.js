import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App container', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('App', () => {
  it('App renders without crashing', () => {
    shallow(<App />);
  });

  it('should render a card title by default', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('h1').text()).toBeTruthy();
  });
});
