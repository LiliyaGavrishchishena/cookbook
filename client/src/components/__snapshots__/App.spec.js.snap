// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container should render 1`] = `ShallowWrapper {}`;
